app/components/banner/Banner.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View, StyleSheet, type ViewStyle, type StyleProp} from 'react-native';
+import {GestureDetector} from 'react-native-gesture-handler';
+import Animated from 'react-native-reanimated';
+
+import {useBannerAnimation} from './hooks/useBannerAnimation';
+import {useBannerGesture} from './hooks/useBannerGesture';
+
+/**
+ * Position where the banner should appear on screen
+ */
+export type BannerPosition = 'top' | 'bottom';
```

Comment:
is this top or bottom of the screen? cause what I see is the offset to be set at -50 or 50, but this is from x=0, y=0 no?

Reply:
BannerPosition refers to where on the screen the banner appears ('top' = top of screen, 'bottom' = bottom of screen). The offset values (-50 and 50) are animation slide-in offsets using translateY, which is relative to the component's rendered position, not screen coordinates. When position='top', the offset is -50 (slides in from above), and when position='bottom', the offset is 50 (slides in from below). The banner starts at translateY = offset (offscreen) and animates to translateY = 0 (its natural position on screen), so the direction makes sense - top banners slide down into view from above, bottom banners slide up into view from below.